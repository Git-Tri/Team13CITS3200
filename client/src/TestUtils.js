const trueErrorLog = console.error;

/**
 * okay this is a weird one to explain
 * basically there an issue with our dev setup (web-pack/react/create react app)
 * where if you test when a component fails it will spam the error log with
 * a message telling you the component failed 
 * since we are testing that, this is completely unneccessary
 * as such i am filtering the console.error to exclude these messages 
 * to give a more readable test output
 * using monkey patching: https://davidwalsh.name/monkey-patching
 */
export function filterConsoleError()
{

    
    console.error = (e) => 
    {

        
        if(typeof(e) === "string" && ! e.includes("React will try to recreate this component tree from scratch using the error boundary you provided"))
        {

            trueErrorLog(e);
        
        }
        else if(typeof(e) !== "string")
        {

            trueErrorLog(e);

        }
        
    }

}

export function unfilterConsoleError()
{

    console.error = trueErrorLog;

}