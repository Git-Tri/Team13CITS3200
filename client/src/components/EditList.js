import React, { Component } from 'react';
import PageHeader from './PageHeader.js';
import EditListTable from "./EditListTable";
import {bindEdit, bindUnstructureData, bindStructuredData} from "../Databinding";
import { Button, Loader, Message, Container} from 'semantic-ui-react';
import { withRouter } from 'react-router-dom';


export class EditList extends Component {

	constructor(props)
	{

		super(props)

		this.state = {data: [],target:[], isLoaded: false, isError: false}

	}

     /**
     * Handles any errors cuased by a sub-component 
     * @param {*} error the error recieved
     * @param {*} errorInfo information about the error
     */
    componentDidCatch(error, errorInfo)
    {

        this.setState({isError: true});

    }


	/**
     * Loads all edits and bind the parsed json to edit objects
     */
    loadData()
    {

        fetch("/editList")
            .then(res => res.json())
            .then(result => 
                {

					let editList = result.editList.map((d) => bindEdit(d));

                    let unstructuredData = result.unstructuredData.map((u) => bindUnstructureData(u))

                    let structuredData = result.structuredData.map((u) => bindStructuredData(u)); 

                    let allData = structuredData.concat(unstructuredData);

                    let target = [];

                    editList.forEach((e,index) => target[index] = allData
                            .find((i) => i.id === e.structuredDataID || i.id === e.unstructuredDataID))

					this.setState({data:editList,isLoaded: true, isError: false,target: target})
                })
            .catch(err => this.setState({isError: true}));

    }

    /**
     * loads the data if it is not already loaded
     */
	loadIfNotAlready()
	{

		if(this.state.isLoaded === false)
		{

			this.loadData();

		}


	}

    /**
     * renders the page in loading state 
     */
	renderLoaded()
	{

		return (
				<div>
					<p> A list of all edits. Click on an edit to view it in detail </p>				
					<div id="container" style={{height:"100vh"}}>
						<EditListTable onSelect={this.routeToEdit.bind(this)} items={{edits:this.state.data,targets:this.state.target}}/>
					</div>
				</div>
		);

	}

	 /**
     * Renders the page if an error has occured
     */
    renderError()
    {

        return(  
        <Message negative>
            <Message.Header>An error has occured</Message.Header>
            <p>Failed to get data from the server.</p>
            <Button color="red" onClick={this.loadData.bind(this)}>Try Again?</Button>
          </Message>
          )

    }

    /**
     * navigates to an edit page with parameters given by inputted edit
     * @param {*} edit the edit inputted 
     */
	routeToEdit(edit)
	{

		this.props.history.push("/add_edit?id=" + edit.editID + "&isbackable=true");
		
	}

    /**
     * renders the page if the data is still loading 
     */
    renderLoading()
    {

        return(<Loader>Loading Data</Loader>)

    }
    
    /**
     * Chooses which render that should be rendered based on 
     * state of the object 
     */
    executeRender()
    {

        if(this.state.isError)
        {

            return this.renderError();

        }
        else if(this.state.isLoaded)
        {

            return this.renderLoaded();

        }
        else
        {

            return this.renderLoading();

        }

    }

	

    /**
     * renders the page 
     */
	render() {

		this.loadIfNotAlready();

		return (<div className="page">
			<PageHeader 
				header={"Edit List"}
				sidebarVisible={this.props.sidebarVisible}
				handleSidebarClick={this.props.handleSidebarClick}
			/>
			<br/>
			<Container>
			<div id="container" style={{minHeight:"100vh"}}>
				{this.executeRender()}
			</div>
            </Container>	
		</div>)
		
	}
}

export default withRouter(EditList);