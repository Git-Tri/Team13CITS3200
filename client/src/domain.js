/**
 * A class representing unstructured data
 * for list you may want to leave the data field undefined
 * NOTE: this code is duplicated on client and server so is only tested on the server
 */
export class UnstructuredData
{

    constructor(id,matchid,title,author,url,published,extracted,data)
    {

        this.id = id;
        this.matchid = matchid;
        this.title = title;
        this.author = author;
        this.url = url;
        this.published = published;
        this.extracted = extracted;
        this.data = data; 

    }

}

/**
 * A class representing structured data, for lists you may want to leave data undefined
 */
export class StructuredData
{

    constructor(id,date,home,away,competitionID,competitionName,data)
    {

        this.id = id;
        this.date = date;
        this.home = home;
        this.away = away;
        this.competitionID = competitionID;
        this.competitionName = competitionName;
        this.data = data; 

    }

}

/**
 * a class representing a match 
 */
export class Match
{

    constructor(id,date,home,away,competitionID,competitionName)
    {

        this.id = id;
        this.date = date;
        this.home = home;
        this.away = away;
        this.competitionID = competitionID;
        this.competitionName = competitionName;

    }

}

class InsertMatch
{

    constructor(id,competitionID,date,home,away,data)
    {

        this.id = id;
        this.competitionID = competitionID;
        this.date = date;
        this.home = home;
        this.away = away;
        this.data = data;
          

    }

}

/**
 * A class representing an edit 
 */
export class Edit 
{

    constructor(editID,structuredDataID,unstructuredDataID,isCorpus,settings,replace,replaceWith,type,order)
    {

        this.editID = editID;
        this.structuredDataID = structuredDataID;
        this.unstructuredDataID = unstructuredDataID;
        this.isCorpus = isCorpus;
        this.settings = settings;
        this.replace = replace;
        this.replaceWith = replaceWith;
        this.type = type;
        this.order = order;
        
    }

}


export class Competition
{

    constructor(id,name,countryName,countryId)
    {

        this.id = id; 
        this.name = name; 
        this.countryName = countryName;
        this.countryId = countryId

    }

}

export class ImportRequest
{

    constructor(begin,end,compId)
    {

        this.begin = begin; //optional

        this.end = end; //optional

        this.compId = compId; //mandatory 

    }

}


export class SearchRequest
{

    constructor(type,value,field)
    {

        this.type = type;
        this.value = value;
        this.field = field; 

    }

}


export class User
{

    constructor(id, username, hash, admin, regkey, token, apikey)
    {

        this.id = id; 
        this.username = username; 
        this.hash = hash;
        this.admin = admin;
        this.regkey = regkey;
        this.token = token;
        this.apikey = apikey;

    }

}