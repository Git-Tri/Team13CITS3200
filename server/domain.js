/**
 * A class representing unstructured data
 * for list you may want to leave the data field undefined
 * NOTE: this code is duplicated on client and server so is only tested on the server
 */
class UnstructuredData
{

    constructor(id,matchid,title,author,url,published,extracted,data)
    {

        this.id = id;
        this.matchid = matchid;
        this.title = title;
        this.author = author;
        this.url = url;
        this.published = published;
        this.extracted = extracted;
        this.data = data; 

    }

}

/**
 * A class representing structured data, for lists you may want to leave data undefined
 */
class StructuredData
{

    constructor(id,date,home,away,competitionID,competitionName,data)
    {

        this.id = id;
        this.date = date;
        this.home = home;
        this.away = away;
        this.competitionID = competitionID;
        this.competitionName = competitionName;
        this.data = data; 

    }

}

/**
 * a class representing a match 
 */
class Match
{

    constructor(id,date,home,away,competitionID,competitionName)
    {

        this.id = id;
        this.date = date;
        this.home = home;
        this.away = away;
        this.competitionID = competitionID;
        this.competitionName = competitionName;

    }

}

class InsertMatch
{

    constructor(id,competitionID,date,home,away,data)
    {

        this.id = id;
        this.competitionID = competitionID;
        this.date = date;
        this.home = home;
        this.away = away;
        this.data = data;
          

    }

}

/**
 * A class representing an edit 
 */
class Edit 
{

    constructor(editID,structuredDataID,unstructuredDataID,isCorpus,settings,replace,replaceWith,type)
    {

        this.editID = editID;
        this.structuredDataID = structuredDataID;
        this.unstructuredDataID = unstructuredDataID;
        this.isCorpus = isCorpus;
        this.settings = settings;
        this.replace = replace;
        this.replaceWith = replaceWith;
        this.type = type;
        
    }

}

class Competition
{

    constructor(id,name,countryName,countryId)
    {

        this.id = id; 
        this.name = name; 
        this.countryName = countryName;
        this.countryId = countryId

    }

}

class User
{

    constructor(id, username, hash, admin, regkey)
    {

        this.id = id; 
        this.username = username; 
        this.hash = hash;
        this.admin = admin;
        this.regkey = regkey;

    }

}


class ImportRequest
{

    constructor(begin,end,compId)
    {

        this.begin = begin; //optional

        this.end = end; //optional

        if(typeof(compId !== "string"))
        {

            throw new Error("Competition must exist and of type string");

        }

        this.compId = compId; //mandatory 

    }

}

class SearchRequest
{

    constructor(type,value,field)
    {

        this.type = type;
        this.value = value;
        this.field = field; 

    }

}

//export all classes 
module.exports = 
{

    UnstructuredData,StructuredData,Match,Edit,Competition,ImportRequest, InsertMatch, SearchRequest, User

}