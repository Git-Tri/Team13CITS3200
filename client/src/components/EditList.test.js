import {EditList} from "./EditList";
import renderer from 'react-test-renderer';
import React from 'react';
import {Edit} from "../domain";
import {filterConsoleError,unfilterConsoleError} from "../TestUtils";
import {shallow, mount} from "enzyme";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("Edit List Tests",function()
{

    
    const testData = [new Edit(1,1,null,false,{},"apple","banna","replace"),
        new Edit(2,null,1,false,{},"football","soccer","replace"),
        new Edit(3,1,null,false,{field:"player"},"gary jones","player 1","replacewithfield"),
        new Edit(4,null,null,true,{},"reece","ryan","replace")]



    const loadDataMock = EditList.prototype.loadData = jest.fn();

    const routeMock = EditList.prototype.routeToEdit = jest.fn();

    beforeAll(() =>
        {

            return new Promise(function(resolve,reject)
            {

                filterConsoleError();               

                resolve();

            })

        });

    afterEach(() => 
    {

        loadDataMock.mockClear();

        routeMock.mockClear();

    })

    test("should render correctly on first load",function()
    {

        expect(renderer.create(<EditList />)
            .toJSON())
            .toMatchSnapshot();

    });

    test("should call load data when component renders",function()
    {
        
        mount(<EditList />);

        expect(loadDataMock).toHaveBeenCalledTimes(1);

    });

    test("should call route on selection",function()
    {

        let component = mount(<EditList />);

        component.instance().routeToEdit();
        
        expect(routeMock).toHaveBeenCalledTimes(1);

    });

    test("should render Error correctly",function(done)
    {

        let component = mount(<EditList />);

        component.setState({isError: true},() => 
        {

            expect(toJson(component)).toMatchSnapshot();

            done();

        })


    });

    test("should render loaded data correctly",function(done)
    {

        let component = mount(<EditList />);

        component.setState({isLoaded: true, data: testData},() => 
        {

            expect(toJson(component)).toMatchSnapshot();

            done();

        })


    });

    test("should call route function when click on cell",function(done)
    {
        
        let component = mount(<EditList />);

        component.setState({isLoaded: true, data: testData},() => 
        {

            component.find("EditRow").first().simulate("click");

            expect(routeMock).toHaveBeenCalledTimes(1);  

            done();

        })



    });

    test("should render loading correctly",function()
    {

        let component = mount(<EditList />);
     
        expect(toJson(component)).toMatchSnapshot();   

    });
  

    test("should render error message when sub-component fails",function(done)
    {

        
        let component = mount(<EditList />);

        let brokenData = testData.slice();

        brokenData.push(undefined);

        component.setState({isLoaded: true, data: brokenData},() => 
        {

            expect(toJson(component)).toMatchSnapshot();

            done();

        })

    })

    afterAll(() =>
    {

        return new Promise(function(resolve,reject)
        {

            unfilterConsoleError();

            resolve();

        })

    });

})