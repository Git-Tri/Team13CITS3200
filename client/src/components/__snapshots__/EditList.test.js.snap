// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit List Tests should render Error correctly 1`] = `
<EditList>
  <div
    className="page"
  >
    <PageHeader
      header="Edit List"
    >
      <div>
        <Button
          as="button"
        >
          <Ref
            innerRef={
              Object {
                "current": <button
                  class="ui button"
                >
                  Show sidebar
                </button>,
              }
            }
          >
            <RefFindNode
              innerRef={
                Object {
                  "current": <button
                    class="ui button"
                  >
                    Show sidebar
                  </button>,
                }
              }
            >
              <button
                className="ui button"
                onClick={[Function]}
              >
                Show sidebar
              </button>
            </RefFindNode>
          </Ref>
        </Button>
        <Header>
          <div
            className="ui header"
          >
            Edit List
          </div>
        </Header>
      </div>
    </PageHeader>
    <br />
    <div
      id="container"
      style={
        Object {
          "height": "100vh",
        }
      }
    >
      <Message
        negative={true}
      >
        <div
          className="ui negative message"
        >
          <MessageHeader>
            <div
              className="header"
            >
              An error has occured
            </div>
          </MessageHeader>
          <p>
            Failed to get data from the server.
          </p>
          <Button
            as="button"
            color="red"
            onClick={[Function]}
          >
            <Ref
              innerRef={
                Object {
                  "current": <button
                    class="ui red button"
                  >
                    Try Again?
                  </button>,
                }
              }
            >
              <RefFindNode
                innerRef={
                  Object {
                    "current": <button
                      class="ui red button"
                    >
                      Try Again?
                    </button>,
                  }
                }
              >
                <button
                  className="ui red button"
                  onClick={[Function]}
                >
                  Try Again?
                </button>
              </RefFindNode>
            </Ref>
          </Button>
        </div>
      </Message>
    </div>
  </div>
</EditList>
`;

exports[`Edit List Tests should render correctly on first load 1`] = `
<div
  className="page"
>
  <div>
    <button
      className="ui button"
      onClick={[Function]}
    >
      Show sidebar
    </button>
    <div
      className="ui header"
    >
      Edit List
    </div>
  </div>
  <br />
  <div
    id="container"
    style={
      Object {
        "height": "100vh",
      }
    }
  >
    <div
      className="ui text loader"
    >
      Loading Data
    </div>
  </div>
</div>
`;

exports[`Edit List Tests should render error message when sub-component fails 1`] = `<EditList />`;

exports[`Edit List Tests should render loaded data correctly 1`] = `
<EditList>
  <div
    className="page"
  >
    <PageHeader
      header="Edit List"
    >
      <div>
        <Button
          as="button"
        >
          <Ref
            innerRef={
              Object {
                "current": <button
                  class="ui button"
                >
                  Show sidebar
                </button>,
              }
            }
          >
            <RefFindNode
              innerRef={
                Object {
                  "current": <button
                    class="ui button"
                  >
                    Show sidebar
                  </button>,
                }
              }
            >
              <button
                className="ui button"
                onClick={[Function]}
              >
                Show sidebar
              </button>
            </RefFindNode>
          </Ref>
        </Button>
        <Header>
          <div
            className="ui header"
          >
            Edit List
          </div>
        </Header>
      </div>
    </PageHeader>
    <br />
    <div
      id="container"
      style={
        Object {
          "height": "100vh",
        }
      }
    >
      <div>
        <p>
           A list of all edits. Click on an edit to view it in detail 
        </p>
        <div
          id="container"
          style={
            Object {
              "height": "100vh",
            }
          }
        >
          <EditListTable
            items={
              Array [
                Edit {
                  "editID": 1,
                  "isCorpus": false,
                  "replace": "apple",
                  "replaceWith": "banna",
                  "settings": Object {},
                  "structuredDataID": 1,
                  "type": "replace",
                  "unstructuredDataID": null,
                },
                Edit {
                  "editID": 2,
                  "isCorpus": false,
                  "replace": "football",
                  "replaceWith": "soccer",
                  "settings": Object {},
                  "structuredDataID": null,
                  "type": "replace",
                  "unstructuredDataID": 1,
                },
                Edit {
                  "editID": 3,
                  "isCorpus": false,
                  "replace": "gary jones",
                  "replaceWith": "player 1",
                  "settings": Object {
                    "field": "player",
                  },
                  "structuredDataID": 1,
                  "type": "replacewithfield",
                  "unstructuredDataID": null,
                },
                Edit {
                  "editID": 4,
                  "isCorpus": true,
                  "replace": "reece",
                  "replaceWith": "ryan",
                  "settings": Object {},
                  "structuredDataID": null,
                  "type": "replace",
                  "unstructuredDataID": null,
                },
              ]
            }
            onSelect={[Function]}
          >
            <Table
              as="table"
              selectable={true}
              striped={true}
            >
              <table
                className="ui selectable striped table"
              >
                <TableHeader
                  as="thead"
                >
                  <thead
                    className=""
                  >
                    <TableRow
                      as="tr"
                      cellAs="td"
                    >
                      <tr
                        className=""
                      >
                        <TableHeaderCell
                          as="th"
                        >
                          <TableCell
                            as="th"
                            className=""
                          >
                            <th
                              className=""
                            >
                              Target
                            </th>
                          </TableCell>
                        </TableHeaderCell>
                        <TableHeaderCell
                          as="th"
                        >
                          <TableCell
                            as="th"
                            className=""
                          >
                            <th
                              className=""
                            >
                              Type
                            </th>
                          </TableCell>
                        </TableHeaderCell>
                        <TableHeaderCell
                          as="th"
                        >
                          <TableCell
                            as="th"
                            className=""
                          >
                            <th
                              className=""
                            >
                              Summary
                            </th>
                          </TableCell>
                        </TableHeaderCell>
                      </tr>
                    </TableRow>
                  </thead>
                </TableHeader>
                <TableBody
                  as="tbody"
                >
                  <tbody
                    className=""
                  >
                    <EditRow
                      data={
                        Edit {
                          "editID": 1,
                          "isCorpus": false,
                          "replace": "apple",
                          "replaceWith": "banna",
                          "settings": Object {},
                          "structuredDataID": 1,
                          "type": "replace",
                          "unstructuredDataID": null,
                        }
                      }
                      isActive={false}
                      key="1"
                      onSelect={[Function]}
                    >
                      <TableRow
                        active={false}
                        as="tr"
                        cellAs="td"
                        onClick={[Function]}
                      >
                        <tr
                          className=""
                          onClick={[Function]}
                        >
                          <TableCell
                            as="td"
                          >
                            <td
                              className=""
                            >
                              Structured Data: 1
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                          >
                            <td
                              className=""
                            >
                              Replace
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                          >
                            <td
                              className=""
                            >
                              <div>
                                <i>
                                  Replace: 
                                </i>
                                 
                                apple
                                  
                                <i>
                                   - with - 
                                </i>
                                  
                                banna
                              </div>
                            </td>
                          </TableCell>
                        </tr>
                      </TableRow>
                    </EditRow>
                    <EditRow
                      data={
                        Edit {
                          "editID": 2,
                          "isCorpus": false,
                          "replace": "football",
                          "replaceWith": "soccer",
                          "settings": Object {},
                          "structuredDataID": null,
                          "type": "replace",
                          "unstructuredDataID": 1,
                        }
                      }
                      isActive={false}
                      key="2"
                      onSelect={[Function]}
                    >
                      <TableRow
                        active={false}
                        as="tr"
                        cellAs="td"
                        onClick={[Function]}
                      >
                        <tr
                          className=""
                          onClick={[Function]}
                        >
                          <TableCell
                            as="td"
                          >
                            <td
                              className=""
                            >
                              Unstructured Data: 1
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                          >
                            <td
                              className=""
                            >
                              Replace
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                          >
                            <td
                              className=""
                            >
                              <div>
                                <i>
                                  Replace: 
                                </i>
                                 
                                football
                                  
                                <i>
                                   - with - 
                                </i>
                                  
                                soccer
                              </div>
                            </td>
                          </TableCell>
                        </tr>
                      </TableRow>
                    </EditRow>
                    <EditRow
                      data={
                        Edit {
                          "editID": 3,
                          "isCorpus": false,
                          "replace": "gary jones",
                          "replaceWith": "player 1",
                          "settings": Object {
                            "field": "player",
                          },
                          "structuredDataID": 1,
                          "type": "replacewithfield",
                          "unstructuredDataID": null,
                        }
                      }
                      isActive={false}
                      key="3"
                      onSelect={[Function]}
                    >
                      <TableRow
                        active={false}
                        as="tr"
                        cellAs="td"
                        onClick={[Function]}
                      >
                        <tr
                          className=""
                          onClick={[Function]}
                        >
                          <TableCell
                            as="td"
                          >
                            <td
                              className=""
                            >
                              Structured Data: 1
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                          >
                            <td
                              className=""
                            >
                              Replace with Field
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                          >
                            <td
                              className=""
                            >
                              <div>
                                <i>
                                  Replace: 
                                </i>
                                 
                                gary jones
                                  
                                <i>
                                   - with - 
                                </i>
                                  
                                player 1
                                <i>
                                   - on field - 
                                </i>
                                  
                                player
                                 
                              </div>
                            </td>
                          </TableCell>
                        </tr>
                      </TableRow>
                    </EditRow>
                    <EditRow
                      data={
                        Edit {
                          "editID": 4,
                          "isCorpus": true,
                          "replace": "reece",
                          "replaceWith": "ryan",
                          "settings": Object {},
                          "structuredDataID": null,
                          "type": "replace",
                          "unstructuredDataID": null,
                        }
                      }
                      isActive={false}
                      key="4"
                      onSelect={[Function]}
                    >
                      <TableRow
                        active={false}
                        as="tr"
                        cellAs="td"
                        onClick={[Function]}
                      >
                        <tr
                          className=""
                          onClick={[Function]}
                        >
                          <TableCell
                            as="td"
                          >
                            <td
                              className=""
                            >
                              All Corpus
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                          >
                            <td
                              className=""
                            >
                              Replace
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                          >
                            <td
                              className=""
                            >
                              <div>
                                <i>
                                  Replace: 
                                </i>
                                 
                                reece
                                  
                                <i>
                                   - with - 
                                </i>
                                  
                                ryan
                              </div>
                            </td>
                          </TableCell>
                        </tr>
                      </TableRow>
                    </EditRow>
                  </tbody>
                </TableBody>
              </table>
            </Table>
          </EditListTable>
        </div>
      </div>
    </div>
  </div>
</EditList>
`;

exports[`Edit List Tests should render loading correctly 1`] = `
<EditList>
  <div
    className="page"
  >
    <PageHeader
      header="Edit List"
    >
      <div>
        <Button
          as="button"
        >
          <Ref
            innerRef={
              Object {
                "current": <button
                  class="ui button"
                >
                  Show sidebar
                </button>,
              }
            }
          >
            <RefFindNode
              innerRef={
                Object {
                  "current": <button
                    class="ui button"
                  >
                    Show sidebar
                  </button>,
                }
              }
            >
              <button
                className="ui button"
                onClick={[Function]}
              >
                Show sidebar
              </button>
            </RefFindNode>
          </Ref>
        </Button>
        <Header>
          <div
            className="ui header"
          >
            Edit List
          </div>
        </Header>
      </div>
    </PageHeader>
    <br />
    <div
      id="container"
      style={
        Object {
          "height": "100vh",
        }
      }
    >
      <Loader>
        <div
          className="ui text loader"
        >
          Loading Data
        </div>
      </Loader>
    </div>
  </div>
</EditList>
`;
